03-transactions.md

```diff
@@ -120,6 +120,28 @@ If a revoked commitment transaction is published, the other party can spend this
 
     <revocation_sig> 1
 
+##### Leased channel (`option_will_fund`)
+In a leased channel, the `to_local` output that pays the `accepter` node
+is modified so that its CSV is equal to the greater of the
+`to_self_delay` or the `lease_end` - `blockheight`.
+
+    OP_IF
+        # Penalty transaction
+        <revocationpubkey>
+    OP_ELSE
+        MAX(`to_self_delay`, `lease_end` - `blockheight`)
```

Comment:
Do you need to encumber `leasor` (iiuc the counterparty offering the lease?)'s HTLC transactions outputs with OP_MAX too ?

Otherwise I think as a leasor I can send all my balance (minus `max_htlc_value_in_flight_msat`) in an offered HTLC output, of which I'm never going to give you the preimage. At expiration broadcast commitment + HTLC-timeout, wait for the CSV delay, and withdraw the balance to a single-owned output ?

Reply:
Yes, this a hole in the protocol at the moment. We chose not to update the HTLC contracts, as that was a much larger change for the first draft. iiuc it also impacts the lockup for failed payments across the entire payment chain, which seems suboptimal as well.

You can limit your exposure to this today by limiting the max htlc inflight for this channel.

Reply:
Note that LND [already](https://github.com/lightningnetwork/lnd/blob/master/input/script_utils.go) adjusts `to_self` in the HTLC-timeout and HTLC-success transactions for Pool.  This partially plugs the hole, but the lessor can still evade the timelock by routing an HTLC to itself and then goading the lessee into force closing.

The solution I described above should block this case as well, but I'm thinking just having HTLC-success and HTLC-timeout lease locks may be enough for the first protocol.  It would be nice to avoid the more complex HTLC surgery.

Reply:
> We could use the multisig + secondary transaction trick we already use with HTLC-success and HTLC-timeout to avoid cltv_expiry lockup across the payment chain. But yes this would complicate the proposal a bit.

I think I'm going to propose doing exactly this (creating second stage txs for the leasor in their remote's commitment tx HTLC outputs), as it's a complete patch for this hole. It is more surgery, but I think worth it in terms of eradicating a potential hole in the protocol. 

I'll also add HTLC locks on the to-self HTLC second stage txs, as you've suggested (and LND does!)